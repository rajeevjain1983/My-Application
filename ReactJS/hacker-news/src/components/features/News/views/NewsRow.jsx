import React from "react";
import withStyle from "../../../common/hoc/withStyle";
import styles from "../style/NewsRow.style";

const NewsRow = ({
  className,
  data,
  hideButtonHandler,
  upVotesButtonHandler,
}) => {
  return (
    <li className={className}>
      <div className="alignTop col">
        <span className="comments">{data.num_comments}</span>
        <div className="upVotes">
          <span>{data.upVotes}</span>
          <button
            onClick={() => upVotesButtonHandler(data.index)}
            className="link"
          >
            <span className="voteArrow"></span>
          </button>
        </div>
      </div>
      <div className="col wrap">
        <div className="col wrap">
          <a className="marginRight title" target="_new" href={data.url}>
            <span>{data.title}</span>
          </a>
          <span className="marginRight colorGray">{data.linkDomain}</span>
        </div>
        <div className="col wrap">
          <span className="marginRight">{`by ${data.author} `}</span>
          <span className="colorGray">{`${data.created_at}  `}</span>
          <button
            onClick={() => hideButtonHandler(data.index)}
            className="link"
          >
            [ hide ]
          </button>
        </div>
      </div>
    </li>
  );
};

export default withStyle(NewsRow, styles);
