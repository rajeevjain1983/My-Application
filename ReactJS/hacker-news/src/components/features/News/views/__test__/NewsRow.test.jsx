import React from "react";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NewsRow from "../NewsRow";

configure({ adapter: new Adapter() });

describe("<NewsRow", () => {
  it("Render NewsRow view", () => {
    const props = {
      data: {
        url: "url",
        index: 1,
        title: "test",
        points: 10,
        author: "test",
        created_at: new Date("2020-05-26T06:02:13.382Z"),
        num_comments: 10,
        hide: false,
        upVotes: 0,
        linkDomain: "google.com",
      },
      hideButtonHandler: () => {},
      upVotesButtonHandler: () => {},
    };
    const component = shallow(<NewsRow {...props} />);

    expect(component).toMatchSnapshot();
  });
});
