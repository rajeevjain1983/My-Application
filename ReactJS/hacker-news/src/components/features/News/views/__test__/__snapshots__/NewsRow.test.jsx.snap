// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsRow Render NewsRow view 1`] = `
<NewsRow
  className="sc-AxjAm ilgvDr"
  data={
    Object {
      "author": "test",
      "created_at": 2020-05-26T06:02:13.382Z,
      "hide": false,
      "index": 1,
      "linkDomain": "google.com",
      "num_comments": 10,
      "points": 10,
      "title": "test",
      "upVotes": 0,
      "url": "url",
    }
  }
  hideButtonHandler={[Function]}
  upVotesButtonHandler={[Function]}
/>
`;
